src/framework/update/internal/updatescenario.cpp

```diff
@@ -38,42 +38,48 @@ using namespace muse;
 using namespace muse::update;
 using namespace muse::actions;
 
-void UpdateScenario::delayedInit()
+void UpdateScenario::checkForUpdate(bool manual, const CheckForUpdateCompleteCallback& callback)
 {
-    if (configuration()->needCheckForUpdate() && multiInstancesProvider()->instances().size() == 1) {
-        QTimer::singleShot(AUTO_CHECK_UPDATE_INTERVAL, [this]() {
-            doCheckForUpdate(false);
-        });
+    if (isCheckInProgress()) {
+        return;
     }
-}
 
-void UpdateScenario::checkForUpdate()
-{
-    if (isCheckStarted()) {
+    if (manual) {
+        doCheckForUpdate(/*manual*/ true, callback);
+        return;
+    }
+
+    if (!configuration()->needCheckForUpdate() || multiInstancesProvider()->instances().size() != 1) {
```

Comment:
Wouldn't it be preferable to move these checks plus the timer to the caller (i.e. `StartupScenario`), so that `checkForUpdate` becomes just `doCheckForUpdate` and nothing else?

I also feel like we might now have two sources of delay, namely the one second from `AUTO_CHECK_UPDATE_INTERVAL`, and whatever delay mechanism the caller might use.